apiary.apib

```diff
@@ -73,17 +73,47 @@ If you find any bug in our API please use [admin@zonky.cz](mailto:admin@zonky.cz
 
 # OAuth 2 authentication
 
-Zonky users are identified using [OAuth 2.0](http://oauth.net/2/) protocol. At the moment we support only OAuth 2 
+Zonky users are identified using [OAuth 2.0](http://oauth.net/2/) protocol. At the moment we support OAuth 2
 [Resource Owner Password Credentials grant flow](https://tools.ietf.org/html/rfc6749#section-4.3) where user receives an access token and
-a refresh token in exchange for their username and password. 
+a refresh token in exchange for their username and password and [Authorization Code Grant Flow ](https://tools.ietf.org/html/rfc6749#page-24) where
+client app obtains authorization code which is later exchanged for access token
 
 **Acting** on **behalf** of other **users** by supplying their **credentials** is strictly **prohibited**. Applications violating this rule will be banned with no mercy. We may eventually extend our implemenation with [Authorization Code Grant flow](https://tools.ietf.org/html/rfc6749#section-4.1) allowing  3rd party applications to act on behalf of a registered user.
 
 ## Access token and refresh token
-To keep things as secure as possible you have to pay attention to the right usage of OAuth authentication protocol. The right way is to obtain 
+To keep things as secure as possible you have to pay attention to the right usage of OAuth authentication protocol. The right way is to obtain
 an access token and a refresh token with user's credentials *on the first authentication request*. Retrieved access token has limited validity (advertised in the
 response). After its expiration you should retrieve a new one with a refresh token which has longer validity.
 
+## Authorization code grant Flow
+### Client onboarding
+Client application must obtain client credentials before it can use authorization code flow. To start onboarding process please send us email to oauth@zonky.cz
+with following details
+- application name
+- application description
+- requested scopes (list of scopes this app will need to operate)
+- redirect url ( target url where is user redirected after granting authorization code or in case of error) Url is verified for match except for params
```

Comment:
I've seen this with many services, so I understand that this is an accepted practice these days. However, how is a command-line application, most likely running isolated behind a restrictive firewall, supposed to provide a redirect URL? 

I see how this is useful for web services, but for headless clients, I don't see the point. That said, if headless clients can continue using the old mechanism, it doesn't matter.

Reply:
This grant flow can't be used without user logging to zonky app. So you need a web browser to log in. There is option that we can instead of redirecting user display the authorization code and user can type it to some console app. Nevertheless this is currently not implemented. So you need to use username and password.